app/src/main/java/org/p2p/wallet/auth/ui/animationscreen/AnimationProgressFragment.kt

```diff
@@ -0,0 +1,122 @@
+                    R.string.onboarding_loading_creating_title
+                } else {
+                    R.string.onboarding_loading_recovery_title
+                }
+                fragmentManager.beginTransaction()
+                    .setCustomAnimations(R.anim.fade_in, R.anim.fade_out)
+                    .add(
+                        android.R.id.content,
+                        create(loadingTitleRes),
+                        fragmentTag
+                    )
+                    .commit()
+            }
+        }
+
+        fun dismiss(fragmentManager: FragmentManager) {
+            fragmentManager.findFragmentByTag(fragmentTag)?.let { fragment ->
+                fragmentManager.beginTransaction()
+                    .remove(fragment)
+                    .commit()
+                isAnimating = false
+            }
+        }
+    }
+
+    private val binding: FragmentAnimationProgressBinding by viewBinding()
+
+    private val loadingTitleRes: Int by args(ARG_TITLE_RES)
+
+    override val statusBarColor: Int = R.color.bg_lime
+    override val navBarColor: Int = R.color.bg_lime
+
+    private var creationProgressJob: Job? = null
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setLoadingAnimationState(isScreenLoading = true)
+    }
+
+    private fun setLoadingAnimationState(isScreenLoading: Boolean) {
+        with(binding) {
+            TransitionManager.beginDelayedTransition(root)
+            loadingAnimationView.isVisible = isScreenLoading
+            animationView.apply {
+                if (isScreenLoading) {
+                    setSystemBarsColors(statusBarColor, R.color.bg_lime)
+                    startCreationProgressJob()
+                    playAnimation()
+                } else {
+                    setSystemBarsColors(statusBarColor, navBarColor)
+                    creationProgressJob?.cancel()
+                    cancelAnimation()
+                }
+            }
+        }
+    }
+
+    private fun startCreationProgressJob() {
+        creationProgressJob = listOf(
+            TimerState(loadingTitleRes, withProgress = false),
+            TimerState(loadingTitleRes),
+            TimerState(R.string.onboarding_loading_title_2),
+            TimerState(R.string.onboarding_loading_title_3),
+        ).asSequence()
+            .asFlow()
+            .onEach {
+                with(binding) {
+                    textViewCreationTitle.setText(it.titleRes)
+                    textViewCreationMessage.isVisible = !it.withProgress
+                    progressBarCreation.isVisible = it.withProgress
+                }
+                delay(2.seconds.inWholeMilliseconds)
+            }
+            .launchIn(viewLifecycleOwner.lifecycleScope)
+    }
+
+    override fun onDestroyView() {
+        creationProgressJob?.cancel()
+        super.onDestroyView()
+    }
+
+    data class TimerState(
```

Comment:
put classes to the top please

Reply:
I don't why we do this, but all classes should be in a separate file. 
Fragment is about fragment, not state

Reply:
ok will move it out 

it was in fragment as it is not needed in other places so it is like an a private inner class

that used only in this fragment